vm_backtrace.c

```diff
@@ -1641,7 +1647,7 @@ thread_profile_frames(rb_execution_context_t *ec, int start, int limit, VALUE *b
 int
 rb_profile_frames(int start, int limit, VALUE *buff, int *lines)
 {
-    rb_execution_context_t *ec = GET_EC();
+    rb_execution_context_t *ec = rb_current_execution_context(false);
```

Comment:
How about to insert `if (ec == 0) return` here? `thread_profile_frames()` from other path `ec` should be valid.